index.js

```diff
@@ -1,20 +1,54 @@
 const display = document.querySelector('.display')
+let newEntry = true
+const operators = ['+', '-', '*', '/']
+
+function containsOperator(operator) {
+  return display.innerText.includes(operator)
+}
 
 function cleanDisplay() {
-    display.innerText = ""
+  display.innerText = ""
 }
 
-function displayValue(value) {
+function displayValue(value) {    
+  if(operators.some(containsOperator)) {
+    let operatorUsed = display.innerText.split(/\+|\-|\/|\*/)
+    if(operatorUsed[1].includes('.') && value == ".") return
+  } else if(!operators.some(containsOperator)) {
     if(display.innerText.includes('.') && value == ".") return
+  }
+
+  updateDisplay(value)
+}
 
+function updateDisplay(value) {
+  if(newEntry) {
+    display.innerText = value
+    newEntry = false
+  } else {
     display.innerText += value
-    return value
+  }
 }
 
 function displayResult() {
-    let operation = display.innerText
-    let result = eval(operation)
+  let operation = display.innerText
+  let result = eval(operation)
 
+  let correctDecimals = countDecimals(result)
```

Comment:
we can improve this variable's name. what do you think about `decimalPlaces` ?

Reply:
Much better!